### DISCUSSION TITLE
Get results in JSON format?

### DISCUSSION THREAD

**Discussion by Posoroko** [searxng/searxng#1789] [Q&A] [answered]
Hi!
Is there a way I could get the results from a searxng search in a JSON file?

Thanks in advance!

**Comment by return42** [comment, marked as answer]
JSON needs to be activated in the `formats` settings in the `search` section: https://docs.searxng.org/admin/engines/settings.html#search

By example:

```yaml
search:
  ...
  formats:
    - html
    - json
```

In the API use `fromat=json` --> https://docs.searxng.org/dev/search_api.html?highlight=format

**Reply by Posoroko** [reply]
Thanks for your quick reply!
So, it's possible. That's great! Do you know of any (already running) instances that would do it?
Or I should maybe just run my own instance. I just suppose it would be much easier to just use an already running instance...

**Reply by return42** [reply]
> That's great! Do you know of any (already running) instances that would do it?

No, sorry I don't .. may you like to ask on our Matrix channel .. [#searxng:matrix.org](https://matrix.to/#/#searxng:matrix.org)

We did not enable JSON by default, since it is often missused by bots.

> Or I should maybe just run my own instance.

Maintain an instance is a lot of work; the maintenance must be continuous and one must have experience in the operation of WEB services --> If ever possible I recommend to use a public instance.

**Reply by Posoroko** [reply]
OK, thanks! I'll ask on[#searxng:matrix.org.

I have no experience with web services, so I'll keep this solution aside for now.

I just want to build a UI that will let me save my search parameters and organise the results differently. If I could use an already running instance's API to get the results in JSON, that would be awesome.

I have another question: I've read the docs on searxng's syntax for more precise searches. But there doesn't seem to be as many tools as other search engine. For example, google accepts the "inurl" keyword. 
Do you know if all keywords accepted by google is usable on searxng?

**Reply by return42** [reply]
> Do you know if all keywords accepted by google is usable on searxng?

SearXNG itself does not have those keywords, since SearXNG is an aggregator and not all connected engines do support those keywords.

SearXNG pass the entire search term to the engines .. if you know the engine (google) and the keyword-syntax (`inurl:`) you can use it; by example `!google inurl:darmarit.org` (OT: very interesting to see that google stores a lot crap about my domain)

A better example (with language selected):  `!google "inurl:searxng.org" :en`

**Reply by Posoroko** [reply]
oh ok! So I could write specific queries for each search engine. Or if I only select google from the search engine list, I could send requests using it's syntax. That's seems fine for my needs.

Thank you so much for taking the time to answer my questions!

**Comment by JOduMonT** [comment]
> JSON needs to be activated in the `formats` settings in the `search` section: https://docs.searxng.org/admin/engines/settings.html#search

_while this is the top answer in top engine like Google and Bing_
## the link should be corrected by: https://docs.searxng.org/admin/settings/settings_search.html

**Reply by Etrit1** [reply]
Please update your post on here and Reddit if you find a public instance that supports JSON ---- OR RSS.

I'm 1/2 way through the list and haven't find a single one that supports RSS. 

And sadly running any kind of service like this is a beyond my abilities. 

I have found a couple SearX instances that support both RSS and JSON but they both do an annoying thing and have the URL for the result in the result as part of the listing so it makes JSON and RSS feeds look horrible. 

*goes back to trying REST of the public instances for RSS*

I'm debating just contacting people that maintain instances and asking them to enable RSS (And Json)   there IS a contact button on every instance so...

**Reply by Etrit1** [reply]
Oh hey!  And then I JUST now found an instance that supports downloading JSON

So here you go!

https://searx.foss.family

The JSON option is to the right after you search. Click on the arrow to expand the "Download results" and the Json will show. 

Added:  

https://search.mdosch.de/searxng/    supports JSON also.. and RSS.

So now my search is done and I happily go build some RSS feeds. 


..sadly no RSS so my search continues but hey! I helped you so that's cool!  I'll update if I run across any other ones that support JSON just incase that one goes down.

**Reply by vojkovic** [reply]
Hi @Etrit1

I'm the host of https://priv.au

As you know, most public instances disable non-html responses due to the bot attacks. I never realised that anyone might want to ever use it.

Would you be interested if I enable all the other result formats (`csv`,`rss`,`json`), but it will be behind a private API and you'll need to send an authentication token that I will give you.

How many searches do you think you will be making? I think I'll rate limit this API to something like 50 searches every 5 minutes.

Does this sound okay?

**Reply by hyprbased** [reply]
can i sign up for one of these private apis?

**Reply by vojkovic** [reply]
> can i sign up for one of these private apis?

Sure, what's your email? I'll send over a token and instructions on how to use it.

**Reply by seang96** [reply]
@vojkovic I'd like one as well if you could. My purpose is to use my own private instance but wanting the extra privacy of using a public instance(s). I was hoping to aggregate a couple of them so the search load would be spread among them similar the idea of splitting DNS traffic among upstream servers.

Thank you!

**Reply by return42** [reply]
> of using a public instance(s). I was hoping to aggregate a couple of them so the search load would be spread among them similar

Hm :thinking: ... isn't that what bots do? ;-)

**Reply by jeremiah-k** [reply]
> Would you be interested if I enable all the other result formats (`csv`,`rss`,`json`), but it will be behind a private API and you'll need to send an authentication token that I will give you.

I'm very interested in this private api method. I self host for friends and family and I'd guess they might give the URL out to others too, which is fine. I've never had any problems with abuse.  

But now I want to enable json (at least) for personal use with AI agents and have some sort of protection in place for results returned in these formats, while keeping html results open.

Is this process documented somewhere?  Or would you mind sharing how you're doing this?  Thank you.

**Reply by vojkovic** [reply]
> Is this process documented somewhere? Or would you mind sharing how you're doing this? Thank you.

Hi - I have custom patches that I apply to my SearXNG instance, [priv.au](https://priv.au) - the source for the auth is [here](https://github.com/privau/searxng/blob/main/src/auth/auth.py). You could use my fork - there's a docker container, the docs are [here](https://github.com/privau/searxng/blob/main/README.md).

I'm running quite busy currently so I don't have the time to draft a proper PR for this feature and I'm not sure if the maintainers even want such a feature.

**Reply by jeremiah-k** [reply]
> Hi - I have custom patches that I apply to my SearXNG instance, [priv.au](https://priv.au) - the source for the auth is [here](https://github.com/privau/searxng/blob/main/src/auth/auth.py). You could use my fork - there's a docker container, the docs are [here](https://github.com/privau/searxng/blob/main/README.md).
> 
> I'm running quite busy currently so I don't have the time to draft a proper PR for this feature and I'm not sure if the maintainers even want such a feature.

This looks great. tyvm!
